repl/src/main/scala/ammonite/repl/interp/Compiler.scala

```diff
@@ -117,8 +117,45 @@ object Compiler{
             dirDeps: Seq[java.io.File],
             dynamicClasspath: VirtualDirectory,
             evalClassloader: => ClassLoader,
+            pluginClassloader: => ClassLoader,
             shutdownPressy: () => Unit): Compiler = new Compiler{
 
+    val PluginXML = "scalac-plugin.xml"
+
+    lazy val plugins0 = {
+      import scala.collection.JavaConverters._
+      val loader = pluginClassloader
+
+      val plugins = loader
+        .getResources(PluginXML)
+        .asScala
+        .toVector
+        .map { url =>
+          val elem = scala.xml.XML.load(url.openStream())
+          val name = (elem \\ "plugin" \ "name").text
+          val className = (elem \\ "plugin" \ "classname").text
+          name -> className
+        }
+        .filter{
+          case ("acyclic", _) => false // conflicts with AmmonitePlugin
+          case _ => true
+        }
+        .collect { case (name, className) if name.nonEmpty && className.nonEmpty =>
+          name -> (
+            className,
+            try Some(loader.loadClass(className)) catch { case _: ClassNotFoundException => None }
+          )
+        }
+
+      val (found, notFound) = plugins.partition(_._2._2.nonEmpty)
```

Comment:
It seems to me that it would be easier for the fast path to use `collect` on both the `found` and `notFound` cases; you could then do partial destructing within the collect and wouldn't need to ignore the things down below